File: common-api/src/main/java/org/study/common/api/params/ResponseParam.java
package com.gw.api.base.params;

import com.gw.api.base.annonation.NotSign;
import com.gw.api.base.enums.RespCodeEnum;
import com.gw.api.base.utils.RandomUtil;

/**
 * 响应用户请求的参数
 * @author chenyf
 * @date 2018-12-15
 */
public class ResponseParam {
    private String resp_code;
    private String resp_msg;
    private String mch_no;
    private Object data;
    private String rand_str;
    private String sign_type;
    @NotSign
    private String sign;
    @NotSign
    private String sec_key = "";

    public static ResponseParam acceptUnknown(String mchNo){
        ResponseParam responseParam = new ResponseParam();
        responseParam.setResp_code(RespCodeEnum.ACCEPT_UNKNOWN.getCode());
        responseParam.setResp_msg(RespCodeEnum.ACCEPT_UNKNOWN.getMsg());
        responseParam.setMch_no(mchNo);
        responseParam.setSign("");
        responseParam.setRand_str(RandomUtil.get32LenStr());
        return responseParam;
    }

    public static ResponseParam acceptFail(String mchNo){
        ResponseParam responseParam = new ResponseParam();
        responseParam.setResp_code(RespCodeEnum.ACCEPT_FAIL.getCode());
        responseParam.setResp_msg(RespCodeEnum.ACCEPT_FAIL.getMsg());
        responseParam.setMch_no(mchNo);
        responseParam.setSign("");
        responseParam.setRand_str(RandomUtil.get32LenStr());
        return responseParam;
    }

    public String getResp_code() {
        return resp_code;
    }

    public void setResp_code(String resp_code) {
        this.resp_code = resp_code;
    }

    public String getResp_msg() {
        return resp_msg;
    }

    public void setResp_msg(String resp_msg) {
        this.resp_msg = resp_msg;
    }

    public String getMch_no() {
        return mch_no;
    }

    public void setMch_no(String mch_no) {
        this.mch_no = mch_no;
    }

    public Object getData() {
        return data;
    }

    public void setData(Object data) {
        this.data = data;
    }

    public String getRand_str() {
        return rand_str;
    }

    public void setRand_str(String rand_str) {
        this.rand_str = rand_str;
    }

    public String getSign() {
        return sign;
    }

    public void setSign(String sign) {
        this.sign = sign;
    }

    public String getSign_type() {
        return sign_type;
    }

    public void setSign_type(String sign_type) {
        this.sign_type = sign_type;
    }

    public String getSec_key() {
        return sec_key;
    }

    public void setSec_key(String sec_key) {
        this.sec_key = sec_key;
    }
}

File: demo-gateway-phpclient/param/ResponseParam.php
<?php
namespace param;

/**
 * 响应参数类
 * Class ResponseParam
 * @package param
 */
class ResponseParam{
    private $resp_code;
    private $biz_code;
    private $biz_msg;
    private $mch_no;
    private $data;
    private $rand_str;
    private $sign_type;
    private $sign;
    private $sec_key;

    /**
     * 魔术方法，通用的setter方法，可以给所有属性赋值
     * @param $name
     * @param $value
     */
    public function __set($name, $value) {
        $this->$name = $value;
    }


    /**
     * @return mixed
     */
    public function getRespCode()
    {
        return $this->resp_code;
    }

    /**
     * @param mixed $resp_code
     */
    public function setRespCode($resp_code): void
    {
        $this->resp_code = $resp_code;
    }

    /**
     * @return mixed
     */
    public function getBizCode()
    {
        return $this->biz_code;
    }

    /**
     * @param mixed $biz_code
     */
    public function setBizCode($biz_code): void
    {
        $this->biz_code = $biz_code;
    }

    /**
     * @return mixed
     */
    public function getBizMsg()
    {
        return $this->biz_msg;
    }

    /**
     * @param mixed $biz_msg
     */
    public function setBizMsg($biz_msg): void
    {
        $this->biz_msg = $biz_msg;
    }

    /**
     * @return mixed
     */
    public function getMchNo()
    {
        return $this->mch_no;
    }

    /**
     * @param mixed $mch_no
     */
    public function setMchNo($mch_no): void
    {
        $this->mch_no = $mch_no;
    }

    /**
     * @return mixed
     */
    public function getData()
    {
        return $this->data;
    }

    /**
     * @param mixed $data
     */
    public function setData($data): void
    {
        $this->data = $data;
    }

    /**
     * @return mixed
     */
    public function getRandStr()
    {
        return $this->rand_str;
    }

    /**
     * @param mixed $rand_str
     */
    public function setRandStr($rand_str): void
    {
        $this->rand_str = $rand_str;
    }

    /**
     * @return mixed
     */
    public function getSignType()
    {
        return $this->sign_type;
    }

    /**
     * @param mixed $sign_type
     */
    public function setSignType($sign_type): void
    {
        $this->sign_type = $sign_type;
    }

    /**
     * @return mixed
     */
    public function getSign()
    {
        return $this->sign;
    }

    /**
     * @param mixed $sign
     */
    public function setSign($sign): void
    {
        $this->sign = $sign;
    }

    /**
     * @return mixed
     */
    public function getSecKey()
    {
        return $this->sec_key;
    }

    /**
     * @param mixed $sec_key
     */
    public function setSecKey($sec_key): void
    {
        $this->sec_key = $sec_key;
    }


}